### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by vgarvardt**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+language: go
+
+go:
+  - 1.x
+  - 1.7
```
Comment (kieranajp) [inline .travis.yml]: why 1.7?

Comment (vgarvardt) [inline .travis.yml]: copy-paste from janus =) will wait for other comments and fix this together with other issues

Comment (vgarvardt) [inline .travis.yml]: fixed, added 1.8 but some projects may still use 1.7 so I left older version either

File: README.md [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+# hellofresh/stats-go
+
+[![Build Status](https://travis-ci.org/hellofresh/stats-go.svg?branch=master)](https://travis-ci.org/hellofresh/stats-go)
+[![GoDoc](https://godoc.org/github.com/hellofresh/stats-go?status.svg)](https://godoc.org/github.com/hellofresh/stats-go)
+[![Go Report Card](https://goreportcard.com/badge/github.com/hellofresh/stats-go)](https://goreportcard.com/report/github.com/hellofresh/stats-go)
+
+> Generic Stats library written in Go
+
+This is generic stats library that we use in our projects to collects services stats and then create monitoring
```
Comment (kieranajp) [inline README.md]: > This is _a_ generic stats library that we use in our projects to _collect_ _services'_ stats and then create monitoring...

File: README.md [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+dashboards to track activity and problems.
+
+## Key Features
+
+* Two stats backends - `statsd` for production and `log` for development environment
+* Fixed metric sections count for all metrics that allows easy monitoring/alerting setup in `graphana`
```
Comment (kieranajp) [inline README.md]: > _to allow_ easy monitoring...

File: README.md [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+                c.JSON(http.StatusOK, "I'm producing stats!")
+        })
+
+        router.Run(":8080")
+}
+```
+
+#### Generalise resources by type and stripping resource ID
+
+In some cases you do not need to collects metrics for all unique requests, but a single metric for requests of the similar type,
```
Comment (kieranajp) [inline README.md]: > In some cases you do not need to _collect_ metrics for all unique requests, but a single metric for requests of _a_ similar type,

File: README.md [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+e.g. access time to concrete users pages does not matter a lot, but average access time is important.
+`hellofresh/stats-go` allows HTTP Request metric modification and supports IDs filtering out of the box, so
```
Comment (kieranajp) [inline README.md]: > supports _ID_ filtering

File: README.md [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+you can get generic metric `get.users.-id-` instead of thousands metrics like `get.users.1`, `get.users.13`,
```
Comment (kieranajp) [inline README.md]: > thousands _of_ metrics

File: README.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+`get.users.42` etc. that may make your `graphite` suffer from overloading.
+
+To use metric generalisation by second level path ID you can pass `stats.HttpMetricNameAlterCallback` instance to
```
Comment (kieranajp) [inline README.md]: > second level path ID_,_ you can pass _a_ `stats.HttpMetricNameAlterCallback`

File: README.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+`stats.StatsClient.SetHttpMetricCallback()`. Also there is a shortcut function `stats.NewHasIDAtSecondLevelCallback()`
+that generates callback handler for `stats.SectionsTestsMap`, and shortcut function `stats.ParseSectionsTestsMap`,
```
Comment (kieranajp) [inline README.md]: > generates _a_ callback handler

File: README.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+that generates sections test map from string, so you can get this values from config.
```
Comment (kieranajp) [inline README.md]: > so you can get _these_ values from config.

File: README.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+It accepts list of sections with test callback in the following format: `<section>:<test-callback-name>`.
```
Comment (kieranajp) [inline README.md]: > accepts _a_ list

File: README.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+Currently the following test callbacks are implemented:
+
+* `true` - second path level is always treated as ID,
+  e.g. `/users/13` -> `users.-id-`, `/users/search` -> `users.-id-`, `/users` -> `users.-id-`
+* `numeric` - only numeric second path level is interpreted as ID,
+  e.g. `/users/13` -> `users.-id-`, `/users/search` -> `users.search`
+* `not_empty` - only not empty second path level is interpreted as ID,
+  e.g. `/users/13` -> `users.-id-`, `/users` -> `users.-`
+
+You can register your own test callback functions using `stats.RegisterSectionTest()` function
```
Comment (kieranajp) [inline README.md]: > using _the_ `stats.RegisterSectionTest()` function

**Comment by kieranajp** [comment]
Sorry for the native english speaker spam :D

**Comment by vgarvardt** [comment]
@kieranajp it's cool, my English is better than I expected, fixed everything, thx! =)

**Comment by italolelis** [comment]
👍

**Pull request merged**